lib/v2/douyin/user.js

```diff
@@ -1,59 +1,126 @@
 const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+const { art } = require('@/utils/render');
+const path = require('path');
 const config = require('@/config').value;
+const { fallback, queryToBoolean } = require('@/utils/readable-social');
+const querystring = require('querystring');
+
+const templates = {
+    desc: path.join(__dirname, 'templates/desc.art'),
+    cover: path.join(__dirname, 'templates/cover.art'),
+    embed: path.join(__dirname, 'templates/embed.art'),
+    iframe: path.join(__dirname, 'templates/iframe.art'),
+};
+
+const resolveUrl = (url, tls = true, forceResolve = false) => {
+    if (!url) {
+        return '';
+    }
+    if (url.startsWith('//')) {
+        return (tls ? 'https:' : 'http:') + url;
+    }
+    if (forceResolve && !url.match(/^https?:\/\//)) {
+        return (tls ? 'https://' : 'http://') + url;
+    }
+    return url;
+};
+
+const proxyVideo = (url, proxy) => {
+    if (!(url && proxy)) {
+        return url + '';
+    }
+    if (proxy.includes('?')) {
+        if (!proxy.endsWith('=')) {
+            proxy += '=';
+        }
+        return proxy + encodeURIComponent(url);
+    } else {
+        if (!proxy.endsWith('/')) {
+            proxy += '/';
+        }
+        return proxy + url;
+    }
+};
+
+const getOriginAvatar = (url) =>
+    resolveUrl(url)
+        .replace(/^(.*\.douyinpic\.com\/).*(\/aweme-avatar\/)([^?]*)(\?.*)?$/, '$1origin$2$3')
+        .replace(/~\w+_\d+x\d+/g, '');
 
 module.exports = async (ctx) => {
-    // 获取 uid
-    const { uid } = ctx.params;
-    const url = 'https://www.douyin.com/user/' + uid;
+    const uid = ctx.params.uid;
+    const routeParams = querystring.parse(ctx.params.routeParams);
```

Comment:
The `querystring` API is [considered Legacy](https://www.npmjs.com/package/querystring). While it is still maintained, new code should use the `URLSearchParams` API instead.
docs: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

Alternative: [query-string](https://www.npmjs.com/package/query-string)